### PR TITLE
theCore: code style guide and formatting script

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: style.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# theCore C/C++ code style
+
+Below is the style guide for theCore.
+
+**Instead of manually formatting your code, use [the source formatting script](scripts/srcformat.sh).**
+
+## Whitespaces and intdentation
```
Comment (RostakaGmfun) [inline style.md]: Indentation

File: style.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+   ```cpp
+   template<typename CrazyAllocator>
+   class my_strange_container;
+   ```  
+
+## Comments
+
+ - **Do not leave commented code.**
+ - Prefer not to keep `TODO` in code without correspoding issue.
```
Comment (InnusiG) [inline style.md]: Prefer not keeping*
instead of:
Prefer not to keep

File: style.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ Place issue number next to the `TODO` with little explanation why it is there:
```
Comment (InnusiG) [inline style.md]: with a little*

File: style.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+   ```cpp
+   // TODO #123: replace foo() with something that exposes smaller IRQ latency.
+   foo();
+   ```
+ - Use doxygen for commenting function ptototypes and class methods.
```
Comment (InnusiG) [inline style.md]: commenting the*

File: style.md (hunk 1 above)
Comment (forGGe) [inline style.md]: done

File: style.md (hunk 2 above)
Comment (forGGe) [inline style.md]: done

File: style.md (hunk 3 above)
Comment (forGGe) [inline style.md]: done

File: style.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+ - Use C++ comment lines for doxygen commeting:
```
Comment (forGGe) [inline style.md]: done

File: style.md (hunk 4 above)
Comment (forGGe) [inline style.md]: done

**Pull request merged**